core/src/main/java/org/apache/brooklyn/core/objs/AbstractConfigurationSupportInternal.java

```diff
@@ -53,7 +67,61 @@
     }
 
     @Override
-    public <T> Maybe<T> getNonBlocking(ConfigKey<T> key) {
+    public <T> Maybe<T> getNonBlocking(final ConfigKey<T> key) {
+        if (key instanceof StructuredConfigKey || key instanceof SubElementConfigKey) {
+            return getNonBlockingResolvingStructuredKey(key);
```

Comment:
@neykov I wondered about using the same impl, but the problem is `getNonBlockingResolvingStructuredKey` isn't a great implementation. If the thread executing the resolving task is starved, then it could timeout even though the underlying task was not blocked. In contrast, the other code-path will first check if the value needs resolved in a special way. If it does not, it will just return the value directly. It is therefore more reliable!